persistence.js

```diff
@@ -117,7 +117,18 @@ class RedisPersistence extends CachedPersistence {
   }
 
   _getRetainedKeysCluster (cb) {
-    this._db.keys(ALL_RETAINEDKEYS, cb)
+    // Get keys of all the masters:
+    const masters = this._db.nodes('master')
+    Promise.all(
+      masters
+        .map((node) => node.keys(ALL_RETAINEDKEYS))
+    ).then((keys) => {
+      // keys: [['key1', 'key2'], ['key3', 'key4']]
+      // flatten the array
+      cb(null, keys.reduce((acc, val) => acc.concat(val), []))
```

Comment:
wrap this in a nexttick or the callback might be called twice in case of errors

Reply:
could you explain me why this could be called twice? In case once of the promises fails the `catch` should be thrown, also why should nextick fix that problem?